src/main/java/me/moocar/logbackgelf/TcpTransport.java

```diff
@@ -0,0 +1,96 @@
+package me.moocar.logbackgelf;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class TcpTransport implements Transport {
+    private final InetAddress graylog2ServerAddress;
+    private final int graylog2ServerPort;
+    private final Lock lock = new ReentrantLock();
+    private Socket socket;
+    public TcpTransport(int graylog2ServerPort, InetAddress graylog2ServerAddress) {
+        this.graylog2ServerPort = graylog2ServerPort;
+        this.graylog2ServerAddress = graylog2ServerAddress;
+    }
+
+
+    @Override
+    public void send(final byte[] data) {
```

Comment:
The functionality of the two `send` functions is identical except that the List arity takes a list. To reduce on duplication, how about this send just dispatches to `send(Arrays.aslist(data))`